src/main.rs

```diff
@@ -572,21 +639,27 @@ fn main_recv(
     // time, and if the network is faster the channel will be full all the time.
     let (sender, receiver) = mpsc::sync_channel::<Chunk>(16);
 
+    let total_len: u64 = plan.files.iter().map(|f| f.len).sum();
+
+    let total_bytes_received = Arc::new(parking_lot::Mutex::new(0u64));
```

Comment:
If we’re changing it for the sender thread, let’s make this an `AtomicU64` as well.